gtsfm/pose_slam/pose_slam.py

```diff
@@ -20,6 +20,27 @@
 
 
 class PoseSlam:
+    def __init__(self, use_gt_for_initialization: bool = False) -> None:
+        self._use_gt_for_initialization = use_gt_for_initialization
+
+    def __initialize_poses(
+        self,
+        pose_init_graph: gtsam.NonlinearFactorGraph,
+        gt_wTi_list: Optional[List[Optional[Pose3]]] = None,
+    ) -> gtsam.Values:
+        initial_values = gtsam.Values()
+        if self._use_gt_for_initialization and gt_wTi_list is not None:
+            logger.info("Using GT for initialization")
+            for i, wTi in enumerate(gt_wTi_list):
+                if wTi is None:
+                    continue
+
+                initial_values.insert(i, wTi)
+
+        initial_values = gtsam.InitializePose3.initialize(pose_init_graph, initial_values, useGradient=False)
```

Comment:
No, since this can fail, I think it has to be either or. So this should be in an else block